src/itest/java/uk/gov/companieshouse/insolvency/data/repository/RepositoryITest.java

```diff
@@ -1,38 +1,35 @@
 package uk.gov.companieshouse.insolvency.data.repository;
 
+import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.autoconfigure.mongo.embedded.EmbeddedMongoAutoConfiguration;
 import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
-import org.springframework.test.context.DynamicPropertyRegistry;
-import org.springframework.test.context.DynamicPropertySource;
-import org.testcontainers.containers.MongoDBContainer;
 import org.testcontainers.junit.jupiter.Testcontainers;
-import org.testcontainers.utility.DockerImageName;
+import uk.gov.companieshouse.api.insolvency.InternalCompanyInsolvency;
+import uk.gov.companieshouse.insolvency.data.AbstractMongoConfig;
+import uk.gov.companieshouse.insolvency.data.InsolvencyRepository;
 
 @Testcontainers
 @DataMongoTest(excludeAutoConfiguration = EmbeddedMongoAutoConfiguration.class)
-public class RepositoryITest {
+public class RepositoryITest extends AbstractMongoConfig {
 
-  static final MongoDBContainer mongoDBContainer = new MongoDBContainer(
-      DockerImageName.parse("mongo:4.0.10"));
-
-  @DynamicPropertySource
-  static void setProperties(DynamicPropertyRegistry registry) {
-    registry.add("spring.data.mongodb.uri", mongoDBContainer::getReplicaSetUrl);
-    mongoDBContainer.start();
-  }
+  @Autowired
+  private InsolvencyRepository insolvencyRepository;
 
   @BeforeAll
   static void setup(){
```

Comment:
Do we need to start container again in test?

Reply:
Yes , otherwise we need to provide mongodb db settings in properties file.

Reply:
I think you can make MongoDBContainer as a bean in config and start post instantiation? Also just wondering we have to start container in @DynamicPropertySource? Its nice to have if we have time not a blocker to approve though :)